syft/event/event.go

```diff
@@ -34,4 +34,7 @@ const (
 
 	// AttestationStarted is a partybus event that occurs when starting an SBOM attestation process
 	AttestationStarted partybus.EventType = "syft-attestation-started-event"
+
+	// GenericProgressStarted is a partybus event that occurs when a user wants to display arbitrary progress messages
+	GenericProgressStarted partybus.EventType = "syft-generic-event-started"
```

Comment:
I don't think we're ready for a common generic progress event. I think we should keep this scoped for the cataloger for now, allowing us to know by the event type what the progress belongs under (always the cataloging task).